inception-ui-external-search/src/main/java/de/tudarmstadt/ukp/inception/app/ui/externalsearch/sidebar/ExternalSearchAnnotationSidebar.java

```diff
@@ -0,0 +1,427 @@
+        }
+
+        private void actionSearch(AjaxRequestTarget aTarget, Form aForm)
+        {
+            IModel<String> targetQuery = searchStateModel.getObject().getTargetQuery();
+            selectedResult = null;
+            if (targetQuery.getObject() == null) {
+                targetQuery.setObject("*.*");
+            }
+
+            searchDocuments(targetQuery.getObject());
+
+            searchStateModel.getObject().getDataProvider().searchDocuments(targetQuery.getObject());
+
+            aTarget.add(dataTableContainer);
+        }
+    }
+
+    private void searchDocuments(String aQuery)
+    {
+        results.clear();
+        applicationEventPublisher.get().publishEvent(new ExternalSearchQueryEvent(this,
+            searchStateModel.getObject().getCurrentRepository().getProject(),
+            userRepository.getCurrentUser().getUsername(), aQuery));
+
+        try {
+            for (ExternalSearchResult result : externalSearchService
+                .query(userRepository.getCurrentUser(),
+                    searchStateModel.getObject().getCurrentRepository(), aQuery)) {
+                results.add(result);
+            }
+        }
+        catch (Exception e) {
+            LOG.error("Unable to perform query", e);
+            error("Unable to load data: " + ExceptionUtils.getRootCauseMessage(e));
+        }
+    }
+
+    public class ResultRowView
+        extends Panel
+    {
+        public ResultRowView(String id, long rowNumber, IModel<ExternalSearchResult> model)
+        {
+            super(id, model);
+
+            ExternalSearchResult result = (ExternalSearchResult) getDefaultModelObject();
+
+            String documentTitle = result.getDocumentTitle();
+
+            Whitelist wl = new Whitelist();
+            wl.addTags("em");
+            Document dirty = Jsoup.parseBodyFragment(result.getHighlights().get(0), "");
+            Cleaner cleaner = new Cleaner(wl);
+            Document clean = cleaner.clean(dirty);
+            clean.select("em").tagName("mark");
+            String highlight = clean.body().html();
+
+            boolean existsSourceDocument = documentService
+                .existsSourceDocument(project, documentTitle);
+
+            // Import and open annotation
+            LambdaAjaxLink link;
+            if (!existsSourceDocument) {
+                link = new LambdaAjaxLink("docLink", _target -> {
+                    selectedResult = result;
+                    try {
+                        documentImporter
+                            .importDocumentFromDocumentRepository(result.getDocumentTitle(),
+                                searchStateModel.getObject().getCurrentRepository());
+
+                        getAnnotationPage().actionShowSelectedDocument(_target,
+                            documentService.getSourceDocument(project, documentTitle));
+                    }
+                    catch (IOException e) {
+                        LOG.error(e.getMessage(), e);
```

Comment:
`LOG.error(e.getMessage(), e);` there is a risk here that `e.getMessage()` might contain the SLF4J placeholder `{}`. The correct way to achieve what you want would be `LOG.error("{}", e.getMessage(), e);`

Reply:
Fixed.